### DISCUSSION TITLE
Forced to restart local web server to save the changes

### DISCUSSION THREAD

**Discussion by arashi-dev** [typegoose/typegoose#636] [Q&A] [answered]
Hello.

I'm forced to restart the local web server every time I edit Models.
I'm using NextJS 12 and Apollo Server playground

**Comment by hasezoey** [comment, marked as answer]
from what i know, this is expected behavior, also what you mean by "forced restarted" (like killing nodemon and restarting it?)

**Reply by arashi-dev** [reply]
yes. I have to kill the terminal and run `npm run dev` command again

**Reply by hasezoey** [reply]
also, could you clarify what you man by `to save the changes`? (what changes? changes to the model / class file?)

**Reply by arashi-dev** [reply]
When I edit the `@Prop` decorator, I can't see the changes until I restart the web server. like default value or `@post` and `@pre` which I am working with them right now. every time I edit the callbacks, I have to restart the web server

**Reply by hasezoey** [reply]
> When I edit the @Prop decorator, I can't see the changes until I restart the web server

i dont know where you got the information that you dont need to restart, but from my knowledge you always needed to

**Reply by arashi-dev** [reply]
no, `npm run dev` in NextJS is in watch mode. every time I save a change in my files, NextJS will refresh the server. but seems it doesn't work well with typegoose

**Reply by hasezoey** [reply]
> every time I save a change in my files, NextJS will refresh the server. but seems it doesn't work well with typegoose

not a problem with typegoose

also, a simple file refresh would not update the models, because mongoose caches them (and typegoose too) so it would need to be removed everywhere and then re-added

**Reply by arashi-dev** [reply]
yup, this can be the reason
Thank you for your time :)
